docs/source/en/model_doc/propainter.md

```diff
@@ -0,0 +1,148 @@
+
+```python
+import av
+import cv2
+import numpy as np
+
+from datasets import load_dataset
+from huggingface_hub import hf_hub_download
+from PIL import Image
+from transformers import ProPainterImageProcessor, ProPainterModel
+
+np.random.seed(0)
+
+def read_video_pyav(container, indices):
+    '''
+    Decode the video with PyAV decoder.
+    Args:
+        container (`av.container.input.InputContainer`): PyAV container.
+        indices (`List[int]`): List of frame indices to decode.
+    Returns:
+        result (np.ndarray): np array of decoded frames of shape (num_frames, height, width, 3).
+    '''
+    frames = []
+    container.seek(0)
+    start_index = indices[0]
+    end_index = indices[-1]
+    for i, frame in enumerate(container.decode(video=0)):
+        if i > end_index:
+            break
+        if i >= start_index and i in indices:
+            frames.append(frame)
+    return np.stack([x.to_ndarray(format="rgb24") for x in frames])
+
+
+def sample_frame_indices(clip_len, frame_sample_rate, seg_len):
+    '''
+    Sample a given number of frame indices from the video.
+    Args:
+        clip_len (`int`): Total number of frames to sample.
+        frame_sample_rate (`int`): Sample every n-th frame.
+        seg_len (`int`): Maximum allowed index of sample's last frame.
+    Returns:
+        indices (`List[int]`): List of sampled frame indices
+    '''
+    converted_len = int(clip_len * frame_sample_rate)
+    end_idx = np.random.randint(converted_len, seg_len)
+    start_idx = end_idx - converted_len
+    indices = np.linspace(start_idx, end_idx, num=clip_len)
+    indices = np.clip(indices, start_idx, end_idx - 1).astype(np.int64)
+    return indices
+
+
+# Using .mp4 files for data:
+
+# video clip consists of 80 frames(both masks and original video) (3 seconds at 24 FPS)
+video_file_path = hf_hub_download(
+    repo_id="ruffy369/propainter-object-removal", filename="object_removal_bmx/bmx.mp4", repo_type="dataset"
+)
+masks_file_path = hf_hub_download(
+    repo_id="ruffy369/propainter-object-removal", filename="object_removal_bmx/bmx_masks.mp4", repo_type="dataset"
+)
+container_video = av.open(video_file_path)
+container_masks = av.open(masks_file_path)
+
+
+# sample 32 frames
+indices = sample_frame_indices(clip_len=32, frame_sample_rate=1, seg_len=container_video.streams.video[0].frames)
+video = read_video_pyav(container=container_video, indices=indices)
+
+masks = read_video_pyav(container=container_masks, indices=indices)
+video = list(video)
+masks = list(masks)
+
+# Forward pass:
+
+device = "cuda"
+image_processor = ProPainterImageProcessor()
+inputs = image_processor(images = video, masks = masks)to(device)
+
+model = transformers.ProPainterModel.from_pretrained("ruffy369/ProPainter").to(device)
+
+# The first input in this always has a value for inference as its not utilised during training
+with torch.no_grad():
+    outputs = model(**inputs)
```

Comment:
Would be great to showcase how the outputs can be visualized

Reply:
done